common/src/main/java/org/valkyrienskies/mod/mixin/client/renderer/entity/MixinEntityRenderDispatcher.java

```diff
@@ -0,0 +1,88 @@
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+import org.valkyrienskies.core.api.ClientShip;
+import org.valkyrienskies.mod.common.VSGameUtilsKt;
+import org.valkyrienskies.mod.common.entity.handling.VSEntityManager;
+
+@Mixin(EntityRenderDispatcher.class)
+public class MixinEntityRenderDispatcher {
+
+    @Inject(method = "render",
+        at = @At(
+            value = "INVOKE",
+            target = "Lnet/minecraft/client/renderer/entity/EntityRenderer;render(Lnet/minecraft/world/entity/Entity;FFLcom/mojang/blaze3d/vertex/PoseStack;Lnet/minecraft/client/renderer/MultiBufferSource;I)V",
+            shift = At.Shift.BEFORE),
+        locals = LocalCapture.CAPTURE_FAILHARD)
+    <T extends Entity> void render(
+        final T entity, final double x, final double y, final double z, final float rotationYaw,
+        final float partialTicks, final PoseStack matrixStack,
+        final MultiBufferSource buffer, final int packedLight, final CallbackInfo ci,
+        final EntityRenderer<T> entityRenderer) {
+
+        final ClientShip ship =
+            (ClientShip) VSGameUtilsKt.getShipObjectManagingPos(entity.level, entity.blockPosition());
+        if (ship != null) {
+            // Remove the earlier applied translation
+            matrixStack.popPose();
+            matrixStack.pushPose();
+
+            VSEntityManager.INSTANCE.getHandler(entity.getType())
+                .applyRenderTransform(ship, entity, entityRenderer, x, y, z,
+                    rotationYaw, partialTicks, matrixStack,
+                    buffer, packedLight);
+        } else if (entity.isPassenger()) {
+            final ClientShip vehicleShip =
+                (ClientShip) VSGameUtilsKt.getShipObjectManagingPos(entity.level,
+                    entity.getVehicle().blockPosition());
+            // If the entity is a passenger and that vehicle is in ship space
+            if (vehicleShip != null) {
+                VSEntityManager.INSTANCE.getHandler(entity.getVehicle().getType())
+                    .applyRenderOnMountedEntity(vehicleShip, entity.getVehicle(), entity, partialTicks, matrixStack);
+            }
+        }
+    }
+
+    @Inject(
+        method = "shouldRender",
+        at = @At("TAIL"),
+        cancellable = true
+    )
+    void shouldRender(final Entity entity, final Frustum frustum,
+        final double camX, final double camY, final double camZ,
+        final CallbackInfoReturnable<Boolean> cir) {
+        if (!cir.getReturnValue()) {
+            final ClientShip ship =
+                (ClientShip) VSGameUtilsKt.getShipObjectManagingPos(entity.level, entity.blockPosition());
+            if (ship != null) {
+                AABB aABB = entity.getBoundingBoxForCulling().inflate(0.5);
```

Comment:
I think theres already a function to transform AABB through a ShipTransform

Reply:
AABBdc#transform

Reply:
Then i need to convert it from mc to joml and back :/, but ig its cleaner this way

Reply:
And would fix scaling